src/app/homepage/dark-mode-toggle/dark-mode-toggle.component.ts

```diff
@@ -0,0 +1,40 @@
+import { DOCUMENT } from '@angular/common';
+import { Component, Inject, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-dark-mode-toggle',
+  templateUrl: './dark-mode-toggle.component.html',
+  styleUrls: ['./dark-mode-toggle.component.scss'],
+})
+export class DarkModeToggleComponent implements OnInit {
+  isDarkMode: boolean;
+
+  private getUserSettingsIsDarkMode(): boolean {
+    return localStorage.getItem('DARK_MODE') === 'true';
+  }
+
+  private setDarkMode(isDarkMode: boolean) {
+    this.isDarkMode = isDarkMode;
+    this.document.documentElement.setAttribute(
+      'mode',
+      isDarkMode ? 'dark' : 'light',
+    );
+  }
+
+  constructor(@Inject(DOCUMENT) private readonly document: Document) {}
+
+  ngOnInit() {
+    const userPrefersDark =
+      window.matchMedia &&
+      window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Using the Media Matcher instead of `window` makes it easier to use Universal.

https://material.angular.io/cdk/layout/overview#mediamatcher